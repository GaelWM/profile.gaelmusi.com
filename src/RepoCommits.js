import React, { Component } from "react";
import Commit from "./Commit";
import CommitList from "./CommitList";
import axios from "axios";

class RepoCommits extends Component {
  constructor() {
    super();
    this.state = {
      commits: []
    };
    this.handleGetRepoCommits = this.handleGetRepoCommits.bind(this);
  }

  handleGetRepoCommits() {
    axios
      .get("https://api.github.com/repos/gaelwm/profile.gaelmusi.com/commits")
      .then(response => this.setState({ commits: response.data }));
  }

  render() {
    const mycommits = this.state.commits.map(function(commit, i) {
      return <Commit index={i} message={commit.commit.message} />;
    });

    return (
      <div className="card-panel">
        <CommitList commits={mycommits} />
        <button className="btn btn-primary" onClick={this.handleGetRepoCommits}>
          Get Last Commits
        </button>
      </div>
    );
  }
}

export default RepoCommits;
