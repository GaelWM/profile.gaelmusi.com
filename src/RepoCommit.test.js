// __tests__/fetch.js
import React from "react";
import {
  render,
  fireEvent,
  cleanup,
  waitForElement
} from "react-testing-library";
// this adds custom jest matchers from jest-dom
//mport "jest-dom/extend-expect";
import axiosMock from "axios"; // the mock lives in a __mocks__ directory
import RepoCommits from "./RepoCommits"; // see the tests for a full implementation

// automatically unmount and cleanup DOM after the test is finished.
afterEach(cleanup);

test("Fetch makes an API call and displays the commits when Get Last Commits is clicked", async () => {
  //ini
  const commits = {
    "0": { index: "1", message: "Change color of a main div" },
    "1": { index: "1", message: "Change file transfer process to sftp" }
  };
  // Arrange
  //axiosMock.get.mockResolvedValueOnce({ commits });
  //const url = "https://api.github.com/repos/gaelwm/profile.gaelmusi.com/commits";

  const spy = jest.fn();
  //const url =
  //"https://api.github.com/repos/gaelwm/profile.gaelmusi.com/commits";
  //const { getByTestId, container } = render(<RepoCommits commits={commits} />);

  const { getByText, getByTestId } = render(<RepoCommits />);

  fireEvent(
    getByText("Get Last Commits"),
    new MouseEvent("click", {
      bubbles: true, // click events must bubble for React to see it
      cancelable: true
    })
  );

  //await waitForElement(() => getByTestId("testRepoCommit"));

  expect(spy).toHaveBeenCalledTimes(1);
  // expect(axiosMock.get).toHaveBeenCalledWith(url);
});
