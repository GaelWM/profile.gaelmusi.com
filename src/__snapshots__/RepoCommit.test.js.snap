// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fetch makes an API call and displays the commits when Get Last Commits is clicked 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card-panel"
        data-testid="testRepoCommit"
      >
        <ol
          class="list-group"
        >
          <li
            class="list-group-item list-group-item-action"
          >
            Change the website backgroung image
          </li>
          <li
            class="list-group-item list-group-item-action"
          >
            Add Login to website
          </li>
        </ol>
        <button
          class="btn btn-primary"
        >
          Get Last Commits
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card-panel"
      data-testid="testRepoCommit"
    >
      <ol
        class="list-group"
      >
        <li
          class="list-group-item list-group-item-action"
        >
          Change the website backgroung image
        </li>
        <li
          class="list-group-item list-group-item-action"
        >
          Add Login to website
        </li>
      </ol>
      <button
        class="btn btn-primary"
      >
        Get Last Commits
      </button>
    </div>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
